### PR TITLE
Update to FileSystem Tests

### PR DESCRIPTION
**Pull request by ianhays**
- Consolidated AccessTime, WriteTime, and CreationTime tests for FileInfo, File, DirectoryInfo, and Directory into one file per folder - GetSetTimes.cs
- Added some Unix tests
- Overhauled the Write/Read/Append tests for the File class and removed many duplicate tests
- Continuation of #2390

### CODE REVIEW

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs [hunk 1]
```diff
@@ -1,462 +1,445 @@
...
+        [InlineData(FileAttributes.ReadOnly)]
+        [InlineData(FileAttributes.Normal)]
+        public void PathAlreadyExistsAsDirectory(FileAttributes attributes)
         {
-            Assert.Throws<DirectoryNotFoundException>(() =>
-            {
-                Directory.CreateDirectory(path);
-            });
+            DirectoryInfo testDir = Create(GetTestFilePath());
+            testDir.Attributes = attributes;
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: What's the purpose of this, to make sure that setting an attribute doesn't change thename of the directory?  If so, should the same thing be done on the corresponding file test?  I think I'm mainly just confused because the PathAlreadyExistsAsFile and PathAlreadyExistsAsDirectory tests appear to be testing very different things even though I'd have thought the only difference was whether the target was a file or a directory.

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs [hunk 2]
```diff
@@ -1,462 +1,445 @@
...
-    }
 
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)] // UNC shares
-    public static void CreateDirectory_UncPathWithoutShareNameAsPath_ThrowsArgumentException()
-    {
-        var paths = IOInputs.GetUncPathsWithoutShareName();
-        foreach (var path in paths)
+        [Fact]
+        public void RootOath()
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: RootPath?

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs [hunk 3]
```diff
@@ -1,462 +1,445 @@
...
-#if !TEST_WINRT // Cannot set current directory to root from appcontainer with it's default ACL
-    /*
-    [Fact]
-    [ActiveIssue(1220)] // SetCurrentDirectory
-    public static void CreateDirectory_DotDotAsPath_WhenCurrentDirectoryIsRoot_DoesNotThrow()
-    {
-        string root = Path.GetPathRoot(Directory.GetCurrentDirectory());
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        public void UnixPathLongerThan256_Throws()
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: It's not that the path is longer than 256, it's that a component of the path is longer than MaxComponent, right?

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs [hunk 4]
```diff
@@ -1,462 +1,445 @@
...
-    }
 
-#if !TEST_WINRT // TODO: Enable once we implement WinRT file attributes
-    [Fact]
-    public static void CreateDirectory_ReadOnlyExistingDirectoryAsPath_DoesNotThrow()
-    {   // DevDivBugs #33833 
-        using (TemporaryDirectory directory = new TemporaryDirectory())
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)] // alternate data streams
+        public void PathWithAlternativeDataStreams_ThrowsNotSupportedException()
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: Nit: alternate, not alternative (I realize this was preexisting your change and you just copied it)

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs [hunk 5]
```diff
@@ -1,462 +1,445 @@
...
+            var paths = IOInputs.GetPathsWithAlternativeDataStreams();
+            Assert.All(paths, (path) =>
+            {
+                Assert.Throws<NotSupportedException>(() => Create(path));
+            });
         }
-    }
 
+        [Fact]
+        [OuterLoop]
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: Why OuterLoop?

File: src/System.IO.FileSystem/tests/Directory/Delete.cs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+        #endregion
+
+        #region UniversalTests
+
+        [Fact]
+        public void ShouldBeAbleToDeleteHiddenDirectory()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            testDir.Attributes = FileAttributes.Hidden;
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/Delete.cs]: Note that this is going to be a nop on Unix, where hidden is defined to mean that the name starts with a period (so setting Hidden would be a rename and thus we don't do it).  If you want to test that a hidden directory can be removed, you should probably have a test that names the directory to begin with a period, verifies (at least on Unix) that it's Hidden, and then deletes it.

File: src/System.IO.FileSystem/tests/Directory/Exists.cs [hunk 7]
```diff
@@ -1,382 +1,287 @@
...
-    }
-
-    [Fact]
-    public static void Exists_PathWithInvalidCharactersAsPath_ReturnsFalse()
-    {
-        var paths = IOInputs.GetPathsWithInvalidCharacters();
 
-        foreach (string path in paths)
+        [Fact]
+        public void PathAlreadyExistsAsFile()
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/Exists.cs]: Why are some of the tests have a name that includes expected return values and some don't?

File: src/System.IO.FileSystem/tests/Directory/Exists.cs [hunk 8]
```diff
@@ -1,382 +1,287 @@
...
+        [Fact]
+        public void DotDotAsPath_ReturnsTrue()
         {
-            PathInfo path = IOServices.GetPath(directory.Path, IOInputs.MaxDirectory, maxComponent: 10);
+            Assert.True(Exists(Path.Combine(TestDirectory, GetTestFileName(), "..")));
+        }
 
-            Directory.CreateDirectory(path.FullPath);
+#if !TEST_WINRT // WinRT cannot access root
+        /*
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/Exists.cs]: In addition to being conditionally compiled in, this is also commented out.  Do we still need this?  Are we missing cases from being tested because this is commented out?

File: src/System.IO.FileSystem/tests/Directory/Exists.cs [hunk 9]
```diff
@@ -1,382 +1,287 @@
...
+#endif
 
-        foreach (string path in paths)
+        [Fact]
+        public void DirectoryLongerThanMaxDirectoryAsPath_DoesntThrow()
         {
-            bool result = Directory.Exists(path);
-
-            Assert.False(result, path);
+            Assert.All((IOInputs.GetPathsLongerThanMaxDirectory()), (path) =>
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/Exists.cs]: Is this relevant to Unix?

File: src/System.IO.FileSystem/tests/DirectoryInfo/CreateSubdirectory.cs [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+
+        [Theory]
+        [InlineData(FileAttributes.Hidden)]
+        [InlineData(FileAttributes.ReadOnly)]
+        [InlineData(FileAttributes.Normal)]
+        public void PathAlreadyExistsAsDirectory(FileAttributes attributes)
+        {
+            string path = GetTestFileName();
+            DirectoryInfo testDir = Directory.CreateDirectory(Path.Combine(TestDirectory, path));
+            testDir.Attributes = attributes;
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/DirectoryInfo/CreateSubdirectory.cs]: Looks like a bunch of these tests are basically the same as the ones I left a few comments on earlier, so I won't duplicate those here.

File: src/System.IO.FileSystem/tests/DirectoryInfo/GetSetTimes.cs [hunk 11]
```diff
@@ -1,119 +1,78 @@
...
+        public delegate DateTime GetTime(DirectoryInfo testDir);
 
-        DirectoryInfo dir = new DirectoryInfo(fileName);
-        dir.Create();
-
-        foreach (TimeProperty timeProperty in Enum.GetValues(typeof(TimeProperty)))
+        public IEnumerable<Tuple<SetTime, GetTime, DateTimeKind>> TimeFunctions()
         {
-            if (!Interop.IsWindows && timeProperty == TimeProperty.CreationTime) // roundtripping birthtime not supported on Unix
+            if (Interop.IsWindows | Interop.IsOSX)
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/DirectoryInfo/GetSetTimes.cs]: Would it make sense to define a set of base properties somewhere where these platform-based definitions can be consolidated, e.g.
```C#
protected static bool SupportsCreationTime { get { return Interop.IsWindows || Interop.IsOSX; } }
```

File: src/System.IO.FileSystem/tests/DirectoryInfo/GetSetTimes.cs [hunk 12]
```diff
@@ -1,119 +1,78 @@
...
-                        if (setUtc)
-                        {
-                            expectedUtc = dt;
-                        }
-                        else
-                        {
-                            expected = dt;
-                        }
-                    }
+            long afterTime = DateTime.Now.AddSeconds(3).Ticks;
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/DirectoryInfo/GetSetTimes.cs]: Ideally I think we'd only capture the UTC times, and then below rather than choosing whether to compare against the local or UTC, we instead decide whether or not to convert the provided time to UTC based on whether it already is.  Math against local times is usually something we try to avoid.

File: src/System.IO.FileSystem/tests/File/GetSetTimes.cs [hunk 13]
```diff
@@ -1,118 +1,129 @@
...
+            {
+                Assert.InRange(tuple.Item2(testFile.FullName).Ticks, tuple.Item3 == DateTimeKind.Local ? beforeTime : utcBeforeTime, tuple.Item3 == DateTimeKind.Local ? afterTime : utcAfterTime);
+            });
         }
 
-        File.Delete(fileName);
+        [OuterLoop]
+        [Fact]
+        [ActiveIssue(1728, PlatformID.AnyUnix)]
+        public void FileDoesntExist_ReturnDefaultValues()
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/File/GetSetTimes.cs]: Should we have a test that is run and passes on Unix for this same condition?

**Comment by stephentoub** [comment]
Thanks, Ian.  I skimmed through it and left some comments, but overall it looks good.  Love seeing all those deletions ;)

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs (hunk 1 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: The original tests were split into four versions of "PathAlreadyExists": File, Directory, HiddenDirectory, ReadOnlyDirectory. I combined the last three tests into one. The tests should function the same otherwise, where the basic structure is:

Create file/directory/hiddendirectory/readonlydirectory
Create new directory
Check that the new directory is the same as the old directory, or catch an exception if the old path referred to a file instead.

Attribute setting is assumed to work correctly for the purposes of this test. You are correct that it is meaning to test preexisting files/directories.

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs (hunk 2 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: Indeed!

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs (hunk 3 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: This one actually <i>is</i> testing MaxPath, not MaxComponent. The MaxComponent in the function call to IOServices.GetPath is given so that GetPath will return a path that is longer than the second variable (257) with no components longer than the third variable (IOInputs.MaxComponent).

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs (hunk 4 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: Fixed in all locations :)

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs (hunk 3 above)
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: I'm still confused.  The name of the method is "UnixPathLongerThan256", but the method is verifying that "subpaths"  which are no longer than MaxComponent all result in PathTooLongExceptions.  What am I missing?

File: src/System.IO.FileSystem/tests/Directory/Exists.cs (hunk 7 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/Directory/Exists.cs]: The naming of my tests is terrible; some are tests I wrote, some are relatively new tests that I left, and others are copied from the comments of the old RunTests methods.

For now I'm just including expected return values if it doesn't seem immediately obvious from the code or from the function calls, but it isn't a great solution. I had planned on fixing them once Sridhar released a concrete test naming guideline to the team, but I fear I may be getting too deep in my inconsistencies to do so easily whenever that happens.

For the next tests I write I'll stick to [platform if applicable][what it's testing] and leave the expected return values for the code, I think.

File: src/System.IO.FileSystem/tests/Directory/Exists.cs (hunk 8 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/Directory/Exists.cs]: This can all probably be removed anyways because it modifies the current directory. I don't know about the WinRT, though.

File: src/System.IO.FileSystem/tests/Directory/Exists.cs (hunk 9 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/Directory/Exists.cs]: No, not really. It isn't testing the same thing; I'll mark it Windows Only.

File: src/System.IO.FileSystem/tests/DirectoryInfo/GetSetTimes.cs (hunk 11 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/DirectoryInfo/GetSetTimes.cs]: Great idea! That would be specifically helpful for Case sensitivity since those checks are all over the place.

File: src/System.IO.FileSystem/tests/File/GetSetTimes.cs (hunk 13 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/File/GetSetTimes.cs]: A test that checks that the Unix implementation throws exceptions, you mean? I think that would be a good idea if we plan on keeping that functionality for a while longer. I mainly left one out because I assumed we would come to a decision on exception/default time soon.

File: src/System.IO.FileSystem/tests/DirectoryInfo/GetSetTimes.cs (hunk 12 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/DirectoryInfo/GetSetTimes.cs]: Ahh okay. Will Fix.

File: src/System.IO.FileSystem/tests/File/GetSetTimes.cs (hunk 13 above)
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/File/GetSetTimes.cs]: >  you mean? 

Yup :)

File: src/System.IO.FileSystem/tests/Directory/Delete.cs (hunk 6 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/Directory/Delete.cs]: Good idea. Will do.

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs (hunk 3 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: It's a bad method that works because of the values of MaxComponent and MaxPath. Subpaths will only have one element that is a fine test against MaxPath, but the iteration of that array is incorrect in the general case. It should look like:

```
DirectoryInfo testDir = Create(GetTestFilePath());
PathInfo path = IOServices.GetPath(testDir.FullName, 257, IOInputs.MaxComponent);
Assert.Throws<PathTooLongException>(() => Create(path.FullPath));
```

**Comment by stephentoub** [comment]
Skimmed the revisions.   LGTM.

**Comment by rqadri133** [comment]
Look at this  http://taskerapp.azurewebsites.net and once you register
click design business process model do you guys want to work together to
create an form expression language (FEL) for this new BPM , we will call it
Fast BPM take a look.


On Mon, Jul 20, 2015 at 5:46 PM, Ian Hays <notifications@github.com> wrote:

>
>    - Consolidated AccessTime, WriteTime, and CreationTime tests for
>    FileInfo, File, DirectoryInfo, and Directory into one file per folder -
>    GetSetTimes.cs
>    - Added some Unix tests
>    - Overhauled the Write/Read/Append tests for the File class and
>    removed many duplicate tests
>    - Continuation of #2390 <https://github.com/dotnet/corefx/pull/2390>
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/dotnet/corefx/pull/2433
> Commit Summary
>
>    - Update to FileSystem Tests
>
> File Changes
>
>    - *D* src/System.IO.FileSystem/tests/Directory/Ansichars.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-0> (46)
>    - *M* src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-1> (651)
>    - *A* src/System.IO.FileSystem/tests/Directory/Delete.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-2> (142)
>    - *D* src/System.IO.FileSystem/tests/Directory/Delete_str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-3> (244)
>    - *D* src/System.IO.FileSystem/tests/Directory/Delete_str_bool.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-4> (291)
>    - *M* src/System.IO.FileSystem/tests/Directory/Exists.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-5> (481)
>    - *D* src/System.IO.FileSystem/tests/Directory/GetCreationTime_str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-6> (141)
>    - *M* src/System.IO.FileSystem/tests/Directory/GetDirectoryRoot_str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-7> (210)
>    - *D* src/System.IO.FileSystem/tests/Directory/GetLastAccessTime_str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-8> (110)
>    - *D* src/System.IO.FileSystem/tests/Directory/GetLastWriteTime_str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-9> (110)
>    - *M* src/System.IO.FileSystem/tests/Directory/GetSetTimes.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-10> (206)
>    - *D* src/System.IO.FileSystem/tests/Directory/IEnumeratorsTests.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-11> (120)
>    - *D* src/System.IO.FileSystem/tests/Directory/Modify_FailSafe.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-12> (582)
>    - *D*
>    src/System.IO.FileSystem/tests/Directory/SetCreationTime_str_dt.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-13> (302)
>    - *D*
>    src/System.IO.FileSystem/tests/Directory/SetLastAccessTime_str_dt.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-14> (291)
>    - *D*
>    src/System.IO.FileSystem/tests/Directory/SetLastWriteTime_str_dt.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-15> (291)
>    - *M* src/System.IO.FileSystem/tests/DirectoryInfo/Create.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-16> (144)
>    - *A*
>    src/System.IO.FileSystem/tests/DirectoryInfo/CreateSubdirectory.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-17> (195)
>    - *D*
>    src/System.IO.FileSystem/tests/DirectoryInfo/CreateSubdirectory_str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-18> (150)
>    - *M* src/System.IO.FileSystem/tests/DirectoryInfo/Delete.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-19> (47)
>    - *D* src/System.IO.FileSystem/tests/DirectoryInfo/Delete_bool.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-20> (107)
>    - *M* src/System.IO.FileSystem/tests/DirectoryInfo/Exists.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-21> (107)
>    - *D* src/System.IO.FileSystem/tests/DirectoryInfo/Extension.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-22> (179)
>    - *M* src/System.IO.FileSystem/tests/DirectoryInfo/GetSetTimes.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-23> (159)
>    - *D*
>    src/System.IO.FileSystem/tests/DirectoryInfo/SetCreationTime_dt.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-24> (250)
>    - *D*
>    src/System.IO.FileSystem/tests/DirectoryInfo/SetLastAccessTime_dt.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-25> (257)
>    - *D*
>    src/System.IO.FileSystem/tests/DirectoryInfo/SetLastWriteTime_dt.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-26> (269)
>    - *D* src/System.IO.FileSystem/tests/DirectoryInfo/ctor_str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-27> (116)
>    - *D* src/System.IO.FileSystem/tests/DirectoryInfo/get_CreationTime.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-28> (136)
>    - *D*
>    src/System.IO.FileSystem/tests/DirectoryInfo/get_LastAccessTime.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-29> (119)
>    - *D* src/System.IO.FileSystem/tests/DirectoryInfo/get_LastWriteTime.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-30> (125)
>    - *A* src/System.IO.FileSystem/tests/File/Append.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-31> (95)
>    - *D* src/System.IO.FileSystem/tests/File/AppendAll_all.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-32> (285)
>    - *D* src/System.IO.FileSystem/tests/File/AppendText_str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-33> (189)
>    - *D* src/System.IO.FileSystem/tests/File/EnumerableAPIs.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-34> (429)
>    - *D* src/System.IO.FileSystem/tests/File/GetCreationTime_str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-35> (152)
>    - *D* src/System.IO.FileSystem/tests/File/GetLastAccessTime_str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-36> (146)
>    - *D* src/System.IO.FileSystem/tests/File/GetLastWriteTime_str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-37> (158)
>    - *M* src/System.IO.FileSystem/tests/File/GetSetTimes.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-38> (207)
>    - *D* src/System.IO.FileSystem/tests/File/ReadAllBytes_Str.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-39> (161)
>    - *D* src/System.IO.FileSystem/tests/File/ReadAll_all.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-40> (474)
>    - *A* src/System.IO.FileSystem/tests/File/ReadWriteAllBytes.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-41> (90)
>    - *A* src/System.IO.FileSystem/tests/File/ReadWriteAllLines.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-42> (160)
>    - *A* src/System.IO.FileSystem/tests/File/ReadWriteAllText.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-43> (136)
>    - *D* src/System.IO.FileSystem/tests/File/SetCreationTime_str_dt.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-44> (295)
>    - *D* src/System.IO.FileSystem/tests/File/SetLastAccessTime_str_dt.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-45> (252)
>    - *D* src/System.IO.FileSystem/tests/File/SetLastWriteTime_str_dt.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-46> (303)
>    - *D* src/System.IO.FileSystem/tests/File/WriteAllBytes_StrBtA.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-47> (194)
>    - *D* src/System.IO.FileSystem/tests/File/WriteAllText.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-48> (532)
>    - *M* src/System.IO.FileSystem/tests/FileInfo/GetSetTimes.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-49> (160)
>    - *D* src/System.IO.FileSystem/tests/FileInfo/get_CreationTime.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-50> (122)
>    - *D* src/System.IO.FileSystem/tests/FileInfo/get_LastAccessTime.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-51> (138)
>    - *D* src/System.IO.FileSystem/tests/FileInfo/get_LastWriteTime.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-52> (45)
>    - *D* src/System.IO.FileSystem/tests/FileInfo/set_CreationTime_dt.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-53> (263)
>    - *M* src/System.IO.FileSystem/tests/PortedCommon/IOInputs.cs
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-54> (10)
>    - *M* src/System.IO.FileSystem/tests/System.IO.FileSystem.Tests.csproj
>    <https://github.com/dotnet/corefx/pull/2433/files#diff-55> (44)
>
> Patch Links:
>
>    - https://github.com/dotnet/corefx/pull/2433.patch
>    - https://github.com/dotnet/corefx/pull/2433.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dotnet/corefx/pull/2433>.
>



-- 






*Rafi QadriSoftware Engineer240-330-7827Email Security Code=00101*
